cpp/trtorchc/main.cpp

```diff
@@ -240,10 +243,17 @@ int main(int argc, char** argv) {
     if (device_type) {
         auto device = args::get(device_type);
         std::transform(device.begin(), device.end(), device.begin(), [](unsigned char c){ return std::tolower(c); });
+	if (gpu_id) {
+	    compile_settings.device.gpu_id = args::get(gpu_id);
+	    cudaSetDevice(compile_settings.device.gpu_id);
```

Comment:
Is it possible to do this in ConversionCtx instead of here since it will not cover any other interface to the compiler?

Reply:
I aligned it with TensorRT's trtexec sample. My concern is it should be handled by the sample application rather than the internal core library. I can still change it otherwise

Reply:
I think in this case we need to handle it since it seems like we provide the bulk of application logic otherwise  the gpu_id doesnt do anything then. If we can do it we can also create an interface that allows people to use pytorch device specifiers to set the device used for compilation. We do need to consider how this will work though. For instance we should probably store the default device so post compilation we return the system back to normal for the user. We would also need to consider how this might work during runtime, if I have a system with 2 different GPUs and I have a program compiled for say TITAN V, what is the user workflow, how would they make sure that the system works properly, I am hesitant to just tell people to use the CUDA API directly. What would be ideal is if we could either configure the system correctly at runtime for the user or give them the options to do that for themselves using TRTorch APIs.

Reply:
Like if I load a TRTorch program maybe in the serialization we store device information so as it is deserialized we search for if there is an applicable GPU and set the right cuda device?

Reply:
I agree we need to handle it internally in that case, If a TRTorch program performs deserialization then it should have the required information about the device as well otherwise, it will be a hassle for end-user. Let me look closer to the serialization/deserialization workflow and try to add device information as well.

Reply:
Do we have a system with 2 different GPUs that we can experiment with? Maybe a DDPX if there isnt an open workstation?

Reply:
I thought you have 2 cards on your ws. We can utilize a DDPX for this. Eventually, it will help us extend the support on Drive AGX as well.

Reply:
Both my cards are TITAN V